dm/syncer/syncer.go

```diff
@@ -3962,3 +3970,26 @@ func (s *Syncer) setGlobalPointByTime(tctx *tcontext.Context, timeStr string) er
 		zap.Any("locationOfTheTime", loc))
 	return nil
 }
+
+func (s *Syncer) getFlushedGlobalPoint() binlog.Location {
+	return s.checkpoint.FlushedGlobalPoint()
+}
+
+func (s *Syncer) getInitExecutedLoc() binlog.Location {
+	s.RLock()
+	defer s.RUnlock()
+	return s.initExecutedLoc.Clone()
+}
+
+func (s *Syncer) initInitExecutedLoc() {
```

Comment:
do we need set `currentLocation` and `startLocation` and `lastLocation` with this func?

Reply:
ExecutedLoc(executed comes from gtid_executed from mysql, this word tells more information than `start` i think) is like the initial start location.

we can init curr/last location later when we need them.